move_base/src/move_base.cpp

```diff
@@ -383,10 +383,45 @@ namespace move_base {
     controller_costmap_ros_->getCostmap()->setConvexPolygonCost(clear_poly, costmap_2d::FREE_SPACE);
   }
 
-  bool MoveBase::clearCostmapsService(std_srvs::Empty::Request &req, std_srvs::Empty::Response &resp){
+  bool MoveBase::clearCostmapsService(std_srvs::Trigger::Request &req, std_srvs::Trigger::Response &resp){
     //clear the costmaps
+    ROS_WARN("Clearing costmaps");
     planner_costmap_ros_->resetLayers();
     controller_costmap_ros_->resetLayers();
+
+    // Make sure the planner costmap is current and updated.
+    float sleep_time = 0.1;
+    float timeout = 1.0;
```

Comment:
I've seen this take longer than 1 second, maybe add some sort of parameter to the trigger service call (timeout?).